### PR TITLE
Improved messaging

### PR DESCRIPTION
**Pull request by cdschmitz**
Messages are now displayed for various events.  The success-messages branch name is misleading because the updates include messages for errors.   This PR also fixes [RE-72](http://capstonedd.cs.pdx.edu:8080/browse/RE-72)

### CODE REVIEW

**Comment by cdschmitz** [comment]
Found a bug, will resubmit later.

**Comment by AaronHampshire** [comment]
So... this looks good. A few design notes. Of the three, I think only the first is really important, and I'm not even sure I'll like it.

(1) I'd have to see it, but I think the messages would look better underneath the header, but above the content. It might "embed" them a bit more, rather than pushing page content out of the way.

(2) In the [Homebuyer Invite](https://drive.google.com/file/d/0B24lwkPmIOELZFBDMEN4Y3pXWXc/view?usp=sharing), it would be nice if there was only one message for both.

(3) When deleting a home or category, it would be nice to get a message, too.

File: RealEstate/apps/core/templates/core/base.html [hunk 1]
```diff
@@ -77,6 +77,12 @@
       </div><!-- /.container-fluid -->
     </nav>
     <div class="container">
+      {% block messages %}
+        <div class="row">
```
Comment (AaronHampshire) [inline RealEstate/apps/core/templates/core/base.html:86]: Not a big deal, but the row class is to set up the bootstrap grid system. You don't have columns here. Not sure it's needed, but it doesn't hurt, either.

File: RealEstate/apps/core/views.py [hunk 2]
```diff
@@ -296,6 +332,16 @@ def post(self, request, *args, **kwargs):
                             content_type="application/json")
 
 
+class PasswordChangeDoneView(BaseView):
+    """
+    This is needed to provide a message to the user that their password change
+    was successful.
+    """
+    def get(self, request, *args, **kwargs):
+        messages.success(request, "Password change successful")
```
Comment (AaronHampshire) [inline RealEstate/apps/core/views.py]: Maybe we can make this more personal? "You successfully changed your password." or "Congratulations! You may have done nothing else with your life, but at least you changed your password."

File: RealEstate/apps/core/views.py [hunk 3]
```diff
@@ -338,7 +384,9 @@ def post(self, request, *args, **kwargs):
                 Realtor.objects.create(user=user)
             user = authenticate(email=email, password=password)
             _login(request, user)
+            messages.success(request, "Realtor welcome message")
```
Comment (AaronHampshire) [inline RealEstate/apps/core/views.py]: I know this is a placeholder, but let's put something workable here just in case we don't get back to it.

**Comment by cdschmitz** [comment]
1)  Currently all the headers are embedded somewhere within the body block. Are you advocating adding a {% block header %} to core/base.html, then pushing {% block messages %} between it and the body block?  Seems reasonable.
2) No problem.
3) This would be easy if delete was not asynchronous, since bootstrap_messages uses server side rendering.  Should be reproducible with javascript.

**Comment by AaronHampshire** [comment]
Let's wait on (1). like you said, it is doable, but I'm not sure if its worth it right now. Especially with Cole's refactor going in soon. 

(3) no worries...

**Comment by cdschmitz** [comment]
(1) is in progress.  Its not a huge change and it does look better.

File: RealEstate/apps/core/views.py (hunk 2 above)
Comment (sanjuktasen) [inline RealEstate/apps/core/views.py]: Chris had a success message which was showing earlier that "you  have successfully change your password" but it was showing on homebuyer dashboard page. I think it should show underneath or top of the password change page.

**Comment by sanjuktasen** [comment]
When a homebuyer wants to save a home with nickname and address, there should be some kind of verification that will check if the nickname and/or address is missing when saving. Now you can just save a home without any nickname and address and it shows nothing on the homebuyer page. I can add the verification if Aaron thinks it is ok.

File: RealEstate/apps/core/views.py (hunk 2 above)
Comment (cdschmitz) [inline RealEstate/apps/core/views.py]: That behavior is intentional. If the password change is successful, it redirects the user back to the home page.

**Comment by cdschmitz** [comment]
Agreed on the verification.  This was brought up originally regarding the Eval/Category views, and the issue stems from the POST data not being passed through form validation.   The issue is significantly more problematic considering the input fields are text boxes instead of just sliders.  We should spin off a separate task for it though.  Good catch!

**Comment by cdschmitz** [comment]
@AaronHampshire I couldnt help myself I just went and added (3).  For now its just deleted objects but the messaging could be extended to other async actions.

**Comment by sjshoemak** [comment]
+1

**Comment by colepalm** [comment]
Looks good as far as I can tell.

**Comment by AaronHampshire** [comment]
Chris, looks good. Go ahead and merge after clearing up the conflicts.
I still wasn't able to see the delete messages, but I don't think it's a huge deal. That was your pet project anyway. :)

**Pull request merged**